### PR TITLE
Task/12010 refactor the getDCCScenarios intro a seperate helper

### PR DESCRIPTION
**Pull request by 30mar**
## Description
refactor the getDCCScenarios intro a seperate helper

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12010

### CODE REVIEW

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/CCLService/CCLScenariosHelper.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	}
+	
+	// MARK: - Private
+
+	private let cclService: CCLServable
+	private let store: HealthCertificateStoring
+
+	// MARK: - Internal
+
+	func showAdmissionScenarios(
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/CCLService/CCLScenariosHelper.swift:23]: There's nothing shown here, is there? I guess this is something like `selectValueViewModelForAdmissionScenarios`? And I think there is nothing async going on here, so you could just `return` the result

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Services/CCLService/CCLScenariosHelper.swift (hunk 1 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Services/CCLService/CCLScenariosHelper.swift]: done

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertificatesTabCoordinator.swift [hunk 2]
```diff
@@ -413,4 +354,40 @@ final class HealthCertificatesTabCoordinator {
 			self.viewController.present(alert, animated: true, completion: nil)
 		}
 	}
+	
+	private func showAdmissionScenarios() {
+		let result = self.cclScenariosHelper.showAdmissionScenarios()
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertificatesTabCoordinator.swift:359]:
```suggestion
		let result = cclScenariosHelper.showAdmissionScenarios()
```

No explicit self needed

File: src/xcode/ENA/ENA/Source/Services/CCLService/CCLScenariosHelper.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	
+	// MARK: - Private
+
+	private let cclService: CCLServable
+	private let store: HealthCertificateStoring
+
+	// MARK: - Internal
+
+	func showAdmissionScenarios() -> Result<SelectValueViewModel, DCCAdmissionCheckScenariosAccessError> {
+		let result = self.cclService.dccAdmissionCheckScenarios()
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/CCLService/CCLScenariosHelper.swift:24]:
```suggestion
		let result = cclService.dccAdmissionCheckScenarios()
```
Same here

File: src/xcode/ENA/ENA/Source/Services/CCLService/CCLScenariosHelper.swift [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	}
+	
+	// MARK: - Private
+
+	private let cclService: CCLServable
+	private let store: HealthCertificateStoring
+
+	// MARK: - Internal
+
+	func showAdmissionScenarios() -> Result<SelectValueViewModel, DCCAdmissionCheckScenariosAccessError> {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/CCLService/CCLScenariosHelper.swift:23]: Still think the naming is off, new suggestion:
```suggestion
	func viewModelForAdmissionScenarios() -> Result<SelectValueViewModel, DCCAdmissionCheckScenariosAccessError> {
```

File: src/xcode/ENA/ENA/Source/Services/CCLService/CCLScenariosHelper.swift [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+			
+	init(cclService: CCLServable, store: HealthCertificateStoring) {
+		self.cclService = cclService
+		self.store = store
+	}
+	
+	// MARK: - Private
+
+	private let cclService: CCLServable
+	private let store: HealthCertificateStoring
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/CCLService/CCLScenariosHelper.swift:19]: Private should come below internal

**Review by nickguendling — CHANGES_REQUESTED** [review]

**Review by nickguendling — APPROVED** [review]

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**